### PR TITLE
[ML] Stop the `frequent_items` aggregation reporting a subset when a superset exists

### PR DESCRIPTION
**Pull request by hendrikmuhs**
depending on order the frequent item set collector reported sets which are sub sets
of sets already collected. Those should not be reported again. This PR implements 
an additional super set check in which case the super set replaces the sub set in the
collector queue

The PR adds test coverage to test certain data distributions, that's how this issue was 
found. It contains already testing code for an upcoming optimization.

### CODE REVIEW

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/bwc

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/bwc

**Comment by hendrikmuhs** [comment]
@elasticmachine update branch

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/part-2

**Comment by hendrikmuhs** [comment]
@elasticmachine update branch

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/part-2

File: docs/changelog/92239.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pr: 92239
+summary: Frequent items sometimes reports sub sets
```
Comment (droberts195) [inline docs/changelog/92239.yaml:2]:
```suggestion
summary: Stop the `frequent_items` aggregation reporting a subset when a superset exists
```

**Review by droberts195 — APPROVED** [review]
LGTM if you could just make the release note clearer

**Pull request merged**